src/log.c

```diff
@@ -480,6 +487,42 @@ static int ensureCapacity(struct raft_log *l)
     return 0;
 }
 
+int logReinstate(struct raft_log *l, raft_term term, unsigned short type)
+{
+    raft_index index;
+    size_t key;
+    struct raft_entry_ref *bucket;
+    struct raft_entry_ref *slot;
+    struct raft_entry *entry;
+
+    if (l->refs_size == 0) {
+        return 0;
+    }
+
+    index = logLastIndex(l) + 1;
+    key = refsKey(index, l->refs_size);
+    bucket = &l->refs[key];
+    if (bucket->count == 0 || bucket->index != index) {
+        return 0;
+    }
+
+    for (slot = bucket; slot != NULL; slot = slot->next) {
+        if (slot->term == term) {
+            slot->count++;
+            l->back++;
+            l->back %= l->size;
+            entry = &l->entries[l->back];
+            entry->term = term;
+            entry->type = type;
+            entry->buf = slot->buf;
```

Comment:
are `buf` and `batch` of the previous entry properly cleaned up when doing this?

Reply:
They are cleaned up when the old entry is truncated away from the logical log, in removeSuffix. At this point in the code those fields are (logically) uninitialized.